aiida/backends/tests/engine/test_work_chain.py

```diff
@@ -636,6 +636,39 @@ def do_run(self):
 
         run_and_check_success(MainWorkChain)
 
+    def test_process_status_sub_processes(self):
+        """Test that process status is set on node when waiting for sub processes."""
+
+        class MainWorkChain(WorkChain):
+
+            @classmethod
+            def define(cls, spec):
+                super(MainWorkChain, cls).define(spec)
+                spec.outline(cls.do_run)
+
+            def do_run(self):
+                pks = []
+                for i in range(2):
+                    node = self.submit(SubWorkChain)
+                    pks.append(node.pk)
+                    self.to_context(subwc=node)
+
+                assert 'Waiting' in self.node.process_status
+                assert str(pks[0]) in self.node.process_status
```

Comment:
Just to double check, this should always work, right? In the sense: SubWorkChain returns immediately, but if tests are run with 1 worker only, the process will have to terminate first to process this function before running the subworkflow? Or since even a single worker can process more messages from RabbitMQ, it is technically possible that the status is not 'Waiting' anymore by the time we get here?

Reply:
No this will always work, but to do so, I did have to put the checks in the same step of the main work chain as where the sub workchains are submitted. The reason is exactly as you expected, since the sub work chains will execute and return straight away, if I try to do it outside, I can only really do it after the `do_run` step, at which point the awaitables have been cleared. Since the message is set as soon as `to_context` is called, I can check straight after.